source/common/network/address_impl.cc

```diff
@@ -130,7 +139,15 @@ int Ipv4Instance::connect(int fd) const {
 }
 
 int Ipv4Instance::socket(SocketType type) const {
-  return ::socket(AF_INET, flagsFromSocketType(type), 0);
+  int fd = ::socket(AF_INET, flagsFromSocketType(type), 0);
+  RELEASE_ASSERT(fd != -1);
+
+#ifdef __APPLE__
+  // Cannot set SOCK_NONBLOCK as a ::socket flag
+  RELEASE_ASSERT(fcntl(fd, F_SETFL, O_NONBLOCK) != -1);
```

Comment:
This overwrites existing flags... It's probably safe, since we've just created the socket, but for my sanity, could you verify that `fcntl(fd, F_GETFL)` returns `0` at this point? Just one-off experiment, not in the PR.